Description: Refactor ntpwait
 This refactors the code a bit to make way for another change.
Forwarded: no
Origin: vendor
Author: Richard Laager <rlaager@wiktel.com>
Last-Update: 2017-11-12
--- a/ntpclients/ntpwait
+++ b/ntpclients/ntpwait
@@ -93,28 +93,22 @@
             time.sleep(sleep)
             continue
 
+        response = repr(session.response)
         if verbose >= 2:
-            sys.stderr.write(repr(session.response) + "\n")
+            sys.stderr.write(response + "\n")
 
         if msg and msg.startswith("***"):
             if verbose:
                 sys.stdout.write("\b" + msg + "\n")
             sys.exit(1)
 
-        m = re.search(r"leap=([^,]*),", repr(session.response))
+        m = re.search(r"leap=([^,]*),", response)
         if m:
             leap = int(m.group(1))
         else:
             sys.stdout.write("\bLeap status not available\n")
             sys.exit(1)
 
-        if leap == ntp.magic.LEAP_NOTINSYNC:
-            if verbose:
-                sys.stdout.write("\b" + "*+:."[i % 4])
-            if i < tries:
-                time.sleep(sleep)
-            continue
-
         if leap in (ntp.magic.LEAP_NOWARNING, ntp.magic.LEAP_ADDSECOND,
                     ntp.magic.LEAP_DELSECOND):
             # We could check "sync" here to make sure we like the source...
@@ -123,8 +117,14 @@
                                  (ntp.util.monoclock() - basetime))
             sys.exit(0)
 
-        sys.stdout.write("\bUnexpected 'leap' status <%s>\n" % leap)
-        sys.exit(1)
+        if leap != ntp.magic.LEAP_NOTINSYNC:
+            sys.stdout.write("\bUnexpected 'leap' status <%s>\n" % leap)
+            sys.exit(1)
+
+        if verbose:
+            sys.stdout.write("\b" + "*+:."[i % 4])
+        if i < tries:
+            time.sleep(sleep)
 
     if verbose:
         sys.stdout.write("\bNo!\nntpd did not synchronize.\n")
